src/mds/agg_mds/adapters.py

```diff
@@ -177,6 +178,102 @@ def getMetadata(self, **kwargs):
         json_data = self.getRemoteDataAsJson(**kwargs)
         return self.normalizeToGen3MDSFields(json_data, **kwargs)
 
+class MPSAdapter(RemoteMetadataAdapter):
+    """
+    Simple adapter for MPS
+    boiler plate taken from IPSRDublin adapter and added MPS-specific needs
+    
+    parameters: filters which currently should be study_ids=id,id,id...
+    """
+
+    @retry(
+        stop=stop_after_attempt(5),
+        retry=retry_if_exception_type(httpx.TimeoutException),
+        wait=wait_random_exponential(multiplier=1, max=10),
+    )
+    def getRemoteDataAsJson(self, **kwargs) -> Tuple[Dict, str]:
+        """ needs to be implemented in derived class """
+        results = {"results": []}
+        if "filters" not in kwargs or kwargs["filters"] is None:
+            return results
+
+        mds_url = kwargs.get("mds_url", None)
+        if mds_url is None:
+            return results
+
+        study_ids = kwargs["filters"].get("study_ids", [])
+
+        if len(study_ids) > 0:
+            for id in study_ids:
+                try:
+                    #get url request put data into datadict
+                    url = f"{mds_url}/{id}/"
+                    response = httpx.get(url)
+                    response.raise_for_status()
+
+                    data_dict = json.loads(response.text)
```

Comment:
I think you can use `response.json()` instead of importing `json` directly

Reply:
@mfshao  You're right -- just changed. Is this ready for a merge or do you want me to do something with the "in development" test script before ready? I was thinking -- for now -- may be best to keep separate from test/agg_mds_adapters.py given how long it is.